### PR TITLE
Upgrade `rust-bitcoin` to `0.31.0`

### LINKED ISSUE #18: Upgrade `rust-bitcoin` to `0.31.0`

### PR DESCRIPTION
**Pull request by jbesraa**
Previous version `0.30.0`.
  
  Resolves #18

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -36,7 +36,10 @@ impl<'a, T: DeserializeParams<'a>> Uri<'a, bitcoin::address::NetworkUnchecked, T
             None => (string, None),
         };
 
-        let address = address.parse().map_err(Error::uri)?;
+        let address = match address.parse() {
+            Ok(address) => address,
+            Err(e) => return Err(Error::Uri(UriError(UriErrorInner::Address(e)))),
+        };
```
Comment (Kixunil) [inline src/de.rs:42]: This looks unnecessary. What's wrong with the previous one?

**Review by Kixunil — COMMENTED** [review]
Thanks! Looks mostly fine but if the `map_err` doesn't work something is broken elsewhere and I want it to be fixed.

File: src/de.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ impl<'a, T: DeserializeParams<'a>> Uri<'a, bitcoin::address::NetworkUnchecked, T
             None => (string, None),
         };
 
-        let address = address.parse().map_err(Error::uri)?;
+        let address = address.parse().map_err(|e| Error::Uri(UriError(UriErrorInner::Address(e))))?;
```
Comment (jbesraa) [inline src/de.rs:39]: not sure whats the root cause of this, but i suspect its something with the new error `ParseError` we are using that is not implementing everything the `Error` is

**Comment by jbesraa** [comment]
@Kixunil thanks for the review.
Fixed-up the commit

File: src/de.rs (hunk 2 above)
Comment (Kixunil) [reply src/de.rs:39]: What? I tried to use `Error::uri` and it works fine.

Comment (jbesraa) [reply src/de.rs:39]: Ha 
Which rust version? 
Can u suggest the code that worked for you?

Comment (Kixunil) [reply src/de.rs:39]: Both 1.48.1 and 1.63, the original line: `let address = address.parse().map_err(Error::uri)?;`

What exact error are you getting? BTW 0.31 raised MSRV to 1.48, so we need to bump that one in README as well.

Comment (jbesraa) [reply src/de.rs]: Hello, thanks for the review!

so for some reason it was not working and after testing it again with @DanGould it seems to be working just fine with the previous line. 
I fixed up a commit to revert this and also added a commit to bump msrv to `1.48`

**Review by DanGould — APPROVED** [review]
tACK with all combinations of features on my machine.

However, `InvalidNetworkError` gives warnings for fields never being read (even on current master). Perhaps this comment should be made into another issue to set those fields to `pub` or provide accessor methods? #[derive(Clone, Debug] on the error is also dead code

**Review by Kixunil — COMMENTED** [review]
Nice, thanks!

**Comment by Kixunil** [comment]
Yeah, let's deal with the warning separately.

**Pull request merged**